javascript/ql/lib/semmle/javascript/Regexp.qll

```diff
@@ -301,6 +301,34 @@ class RegExpAlt extends RegExpTerm, @regexp_alt {
   override string getAPrimaryQlClass() { result = "RegExpAlt" }
 }
 
+/**
+ * An intersection term, that is, a term of the form `[[a]&&[ab]]`.
+ *
+ * Example:
+ *
+ * ```
+ * /[[abc]&&[bcd]]/v - which matches 'b' and 'c' only.
+ * ```
+ */
+class RegExpIntersection extends RegExpTerm, @regexp_intersection {
+  /** Gets an intersected term of this term. */
+  RegExpTerm getIntersectedTerm() { result = this.getAChild() }
+
+  /** Gets the number of intersected terms of this term. */
+  int getNumIntersectedTerm() { result = this.getNumChild() }
+
+  override predicate isNullable() { this.getIntersectedTerm().isNullable() }
+
+  override string getAMatchedString() {
+    exists(string s | s = this.getChild(0).getAMatchedString() |
+      forall(int i | i in [1 .. this.getNumChild() - 1] | s = this.getChild(i).getAMatchedString()) and
+      result = s
+    )
+  }
+
```

Comment:
```suggestion
```
I'd suggest simply removing this due to performance concerns. There are more efficient ways to write this, but that's not something we need to deal with right now.